cli/flox/src/commands/edit.rs

```diff
@@ -180,8 +180,9 @@ impl Edit {
             .map_err(apply_doc_link_for_unsupported_packages);
 
             match result {
-                Err(EnvironmentError::Core(CoreEnvironmentError::LockedManifest(e))) => {
-                    message::error(format_locked_manifest_error(&e));
+                Err(EnvironmentError::Core(e @ CoreEnvironmentError::LockedManifest(_)))
+                | Err(EnvironmentError::Core(e @ CoreEnvironmentError::DeserializeManifest(_))) => {
+                    message::error(format_core_error(&e));
```

Comment:
Just to confirm I understand: the issue with #1502 is that we weren't catching `DeserializeManifest`, so it was hitting `bail!(e)`?

How much effort would it be to add a test? Would it have to be an integration test and using `expect`?

Reply:
shouldnt be needing expect at least
i can look into adding  a unit test of integration test as fallback

Reply:
I was thinking it might have to be because the functions checks for interactive and launches an editor. If they can be stubbed out in a unit test that would be good 👍🏻